client/asset/eth/multirpc.go

```diff
@@ -141,28 +144,31 @@ func (p *provider) failed() bool {
 
 // bestHeader get the best known header from the provider, cached if available,
 // otherwise a new RPC call is made.
-func (p *provider) bestHeader(ctx context.Context, log dex.Logger) (*types.Header, error) {
-	ctx, cancel := context.WithTimeout(ctx, defaultRequestTimeout)
-	defer cancel()
+func (p *provider) bestHeader(ctx context.Context, log dex.Logger, net dex.Network) (*types.Header, error) {
```

Comment:
You could also make `net` a field on `provider`, in which case you would add the argument to `connectProviders`.

Reply:
Doing that.